src/core/mavlink_mission_transfer.cpp

```diff
@@ -767,4 +775,126 @@ void MAVLinkMissionTransfer::ClearWorkItem::callback_and_reset(Result result)
     _done = true;
 }
 
+MAVLinkMissionTransfer::SetCurrentWorkItem::SetCurrentWorkItem(Sender& sender,
+    MAVLinkMessageHandler& message_handler,
+    TimeoutHandler& timeout_handler,
+    uint8_t type,
+    int current,
+    ResultCallback callback) :
+    WorkItem(sender, message_handler, timeout_handler, type),
+    _current(current),
+    _callback(callback)
+{
+    std::lock_guard<std::mutex> lock(_mutex);
+
+    _message_handler.register_one(
+        MAVLINK_MSG_ID_MISSION_CURRENT,
+        [this](const mavlink_message_t& message) { process_mission_current(message); },
+        this);
+
+    _message_handler.register_one(
+        MAVLINK_MSG_ID_STATUSTEXT,
+        [this](const mavlink_message_t& message) { process_status_text(message); },
+        this);
+}
+
+MAVLinkMissionTransfer::SetCurrentWorkItem::~SetCurrentWorkItem()
+{
+    std::lock_guard<std::mutex> lock(_mutex);
+    _message_handler.unregister_all(this);
+    _timeout_handler.remove(_cookie);
+}
+
+void MAVLinkMissionTransfer::SetCurrentWorkItem::start()
+{
+    std::lock_guard<std::mutex> lock(_mutex);
+
+    _started = true;
+
+    // If we coudln't find it, the requested item is out of range and probably an invalid argument.
+    if (_current < 0) {
```

Comment:
We should also check for greater than length of items. Although never mind, we don't have that information in this class, the autopilot will have to take care of it.

Reply:
It was my thought but I did not find a way to check it unless we receive the item in parameters